File: spec/exec_environment/test_player.rb
#! /usr/bin/env ruby

#test_player.rb
#Alex Sjoberg
#1/8/2014
#

#imports
require 'socket'
require 'optparse'

#Parsing command line arguements
$options = {}
OptionParser.new do |opts|
  opts.banner = "Usage: client.rb --name [name] -p [port]"
  
  opts.on('-p' , '--port [PORT]' , 'Port for client to connect to') { |v| $options[:port] = v}
  opts.on('-n' , '--name [NAME]' , 'Name the ref will use to identify the player') { |v| $options[:name] = v}

end.parse!

class TestPlayer

    def initialize()
      hostname = 'localhost'
      port = $options[:port]
      @ref_socket = TCPSocket.open(hostname,port)
      @name = $options[:name]
      @ref_socket.puts(@name)
    end

    #Connects to referee and guess random letter, closes when referee says wins
    def play
      response_options = ['a','b','c','w']
      while input = @ref_socket.gets
        #puts "Got input: "+input
        if input.include?('move')
          blah = response_options.sample
          @ref_socket.puts blah
          #puts "Sent "+blah
        elsif input.include?('wins')
          @ref_socket.close
          return
        end
      end
    end

end

p1 = TestPlayer.new
p1.play

File: spec/exec_environment/test_player.py
#! /usr/bin/env python

#test_player.py
#Douglas Brown
#1/15/2014
#

#imports
from optparse import OptionParser
import socket
from random import choice

#Parsing command line arguments
#Usage: client.py --name [name] -p [port]"
parser = OptionParser()
parser.add_option("-p","--port",action="store",type="int",dest="port")
parser.add_option("-n","--name",action="store",type="string",dest="name")
(options, args) = parser.parse_args()
#print "port"
#print options.port
#print "name"
#print options.name

HOST = 'localhost'
PORT = options.port
NAME = options.name



s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
ip = socket.gethostbyname(HOST)
#print 'Ip address of ' + HOST + ' is ' + ip
s.connect((ip,PORT)) #Connect to server
#print 'Socket Connected to ' + HOST + ' on ip ' + ip

message = str(NAME)+"\n"
s.send(message)   

#Now receive data
while True:
    reply = s.recv(4096)
    #print "Got input: "+reply.strip()
    if "move" in reply:
        guesses = ['a','b','c','w']
        blah = choice(guesses)+"\n"
        s.send(blah)
        #print "Sent "+blah
    elif "wins" in reply:
        break





